extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcRecorder.java

```diff
@@ -165,6 +165,27 @@ public void handle(AsyncResult<OAuth2Auth> event) {
                 });
 
                 auth = cf.join();
+
+                if (!ApplicationType.WEB_APP.equals(oidcConfig.applicationType)) {
+                    if (oidcConfig.token.refreshTimeout.isPresent()) {
+                        throw new RuntimeException(
+                                "The logout path can only be enabled for " + ApplicationType.WEB_APP + " application types");
+                    }
+                    if (oidcConfig.logout.path.isPresent()) {
+                        throw new RuntimeException(
+                                "The logout path can only be enabled for " + ApplicationType.WEB_APP + " application types");
+                    }
+                }
+
+                String endSessionEndpoint = OAuth2AuthProviderImpl.class.cast(auth).getConfig().getLogoutPath();
+
+                if (oidcConfig.logout.path.isPresent()) {
+                    if (endSessionEndpoint == null) {
```

Comment:
@pedroigor Would you like to support the case where no discovery is possible ? We simply add it and check if no discovered path is available. I can do it later though, up to you